File: Data Structures & Algorithms/validate-parentheses/submission-2.cs
public class Solution {
    public bool IsValid(string s) {
        var stack = new Stack<char>();
        foreach (char c in s)
        {
            if (c == '(' || c == '{' || c == '[')
            {
                stack.Push(c);
            }
            else if (stack.Count > 0)
            {
                char prev = stack.Pop();
                if (c == ')' && prev != '(') return false;
                if (c == '}' && prev != '{') return false;
                if (c == ']' && prev != '[') return false;
            }
            else
            {
                return false;
            }
        }
        return stack.Count == 0;
    }
}

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-2.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        if len(tokens) == 0: return 0
        
        operators = ("+", "-", "*", "/")
        stack = []

        for token in tokens:
            if token in operators:
                op2 = stack.pop()
                op1 = stack.pop()
                match token:
                    case "+": res = op1 + op2
                    case "-": res = op1 - op2
                    case "*": res = op1 * op2
                    case "/": res = int(op1 / op2)
                stack.append(res)
            else:
                stack.append(int(token, 10))
        return stack.pop()
